templates/corporate/upgrade.html

```diff
@@ -9,7 +9,13 @@
 <div id="upgrade-page" class="register-account flex full-page">
     <div class="center-block new-style">
         <div class="pitch">
-            <h1>Upgrade {{ realm.name }} to
+            <h1>Upgrade
+                {% if is_self_hosted %}
+                {{ realm.host }}
+                {% else %}
+                {{ realm.name }}
+                {% endif %}
+                to
```

Comment:
I think we want to change the parameters passed into this template so that we just pass a `{{ customer_name }}` parameter, rather than a `realm` object. (In general, we're likely need to not pass objects like `realm` that might have different types in favor of passing the actual data to display).

Reply:
This type of variation is what the `update_context_initial_upgrade` method is intended to do. We can potentially help the type-checker verify our logic by making that function return a `TypedDict` defining what parameters it's supposed to output, which might cause it to complain if different `BillingSession` subclasses return different types... so that we'd explicitly pass parameters like `is_self_hosting: bool` in all of them just to keep it readable.

Reply:
Done.